1. Problem Statement

This dataset provides comprehensive sales information from two distinct stores, namely ExcelMart and MecStore, operating across various states. Each entry records the sales activity of specific merchandise within a particular state and store. The dataset captures the diversity of products offered by each store, reflecting their unique inventory and sales performance across different regions.

2. Data Collection <br>
2.1 Import Data and Required Collection

In [1]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
%matplotlib inline
import warnings
warnings.filterwarnings('ignore')

In [2]:
df_train=pd.read_csv("train.csv")

In [3]:
df_train.head()

row_id        date   state      store      product  num_sold
0     0.0  01-01-2015  Kerala  ExcelMart      Mec Mug     329.0
1     1.0  01-01-2015  Kerala  ExcelMart      Mec Hat     520.0
2     2.0  01-01-2015  Kerala  ExcelMart  Mec Sticker     146.0
3     3.0  01-01-2015  Kerala   MecStore      Mec Mug     572.0
4     4.0  01-01-2015  Kerala   MecStore      Mec Hat     911.0

In [4]:
df_train.shape

(24590, 6)

The dataset contains 24590 rows and 6 columns.

<h2>2.2 Dataset information</h2>
This file serves as the training set and contains detailed sales data, including the date of sale, state, store name, product type, and quantity of items sold. Each row represents a transaction, showcasing the variety of products sold by each store in different states over time. <br>
1. date of the sale <br>
2. State <br>
3. Store name <br>
4. Product type <br>
5. Quantity of items sold

In [5]:
df_test=pd.read_csv("test.csv")

In [6]:
df_test.head

<bound method NDFrame.head of       row_id        date  state      store      product
0      22297  23-05-2018  Delhi  ExcelMart      Mec Hat
1      22298  23-05-2018  Delhi  ExcelMart  Mec Sticker
2      22299  23-05-2018  Delhi   MecStore      Mec Mug
3      22300  23-05-2018  Delhi   MecStore      Mec Hat
4      22301  23-05-2018  Delhi   MecStore  Mec Sticker
...      ...         ...    ...        ...          ...
3995   26293  31-12-2018  Delhi  ExcelMart      Mec Hat
3996   26294  31-12-2018  Delhi  ExcelMart  Mec Sticker
3997   26295  31-12-2018  Delhi   MecStore      Mec Mug
3998   26296  31-12-2018  Delhi   MecStore      Mec Hat
3999   26297  31-12-2018  Delhi   MecStore  Mec Sticker

[4000 rows x 5 columns]>

In [8]:
import pycaret

In [9]:
from pycaret.regression import *

ImportError: cannot import name 'interp' from 'scipy' (E:\01_BTECH CS\datascience\Academic Forecasting System\venv\Lib\site-packages\scipy\__init__.py)